src/aws-elasticache/serverless-cache.ts

```diff
@@ -0,0 +1,509 @@
+      grantee,
+      actions,
+      resourceArns: [
+        Stack.of(this).formatArn({
+          service: 'elasticache',
+          resource: 'serverlesscache',
+          resourceName: this.serverlessCacheName,
+          arnFormat: ArnFormat.COLON_RESOURCE_NAME,
+        }),
+      ],
+    });
+  }
+
+  /**
+   * Permits an IAM principal to perform connect to the serverless cache.
+   *
+   * Actions: Connect
+   *
+   * @param grantee The principal to grant access to.
+   * @see https://docs.aws.amazon.com/AmazonElastiCache/latest/dg/IAM.IdentityBasedPolicies.html#iam-connect-policy
+   */
+  public grantConnect(grantee: aws_iam.IGrantable): aws_iam.Grant {
+    return this.grant(grantee, 'elasticache:Connect');
+  }
+}
+
+/**
+ * Represents a ElastiCache Serverless Cache construct in AWS CDK.
+ *
+ * @example
+ * declare const vpc: aws_ec2.IVpc;
+ *
+ * const serverlessCache = new ServerlessCache(
+ *   stack,
+ *   'ServerlessCache',
+ *   {
+ *     serverlessCacheName: 'my-serverlessCache',
+ *     engine: Engine.VALKEY,
+ *     vpc,
+ *   },
+ * );
+ */
+export class ServerlessCache extends SeverlessCacheBase implements IServerlessCache {
+  /**
+   * The serverless cache ARN
+   */
+  readonly serverlessCacheArn: string;
+  /**
+   * The serverless cache name
+   */
+  readonly serverlessCacheName: string;
+  /**
+   * The DNS hostname of the cache node
+   */
+  readonly endpointAddress: string;
+  /**
+   * The port number that the cache engine is listening on
+   */
+  readonly endpointPort: number;
+  /**
+   * The DNS hostname of the cache node
+   */
+  readonly readerEndpointAddress: string;
+  /**
+   * The port number that the cache engine is listening on
+   */
+  readonly readerEndpointPort: number;
+
+  /**
+   * The connection object associated with the ElastiCache Serverless Cache.
+   */
+  public readonly connections: aws_ec2.Connections;
+
+  private readonly props: ServerlessCacheProps;
+  private readonly securityGroups: aws_ec2.ISecurityGroup[];
+  private readonly vpcSubnets: aws_ec2.SubnetSelection;
+
+  constructor(scope: Construct, id: string, props: ServerlessCacheProps) {
+    super(scope, id, {
+      physicalName:
+        props.serverlessCacheName ??
+        Lazy.string({
+          produce: () => Names.uniqueResourceName(this, { maxLength: 40 }).toLowerCase(),
+        }),
+    });
+    this.props = props;
+
+    this.securityGroups = props.securityGroups ?? [
+      this.createSecurityGroup(this, 'SecurityGroup', {
+        vpc: this.props.vpc,
+        description: 'Automatic generated security group for ElastiCache Serverless Security Group',
+      }),
+    ];
+
+    this.vpcSubnets = props.vpcSubnets ?? {
+      subnetType: aws_ec2.SubnetType.PRIVATE_WITH_EGRESS,
+    };
+
+    this.validateServerlessCacheName();
+    this.validateSnapshotRetentionLimit();
+    this.validateFinalSnapshotName();
+    this.validateUserGroup();
+
+    const serverlessCache = this.createResource(this, 'Resource', {
```

Comment:
[CacheUsageLimits](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticache-serverlesscache.html#cfn-elasticache-serverlesscache-cacheusagelimits) is not exposed yet because the schema in the version of aws-cdk-lib currently used by the open constructs library differs from the latest version. It will be added once the latest version is adopted.